### PR TITLE
Always rebuild it when building dist

### LINKED ISSUE #13372: scons launcher and scons dist don't rebuild
### TL;DR
`scons launcher` or `scons dist` commands don't  seem to be able to pick up code changes after first execution. In other words, they  can build  successfully for the first time, but they  cannot rebuild as they don't appear to see that code has changed.
As to why this can be useful, in my example, I have NVDA build environment on my personal laptop, but I am trying to debug an issue with MS WOrd, which is only installed on my work laptop. I cannot build NVDA on work laptop, nor can I install MS Word on my personal laptop. As a result I need to frequently repeat the cycle of making code change, rebuild and test on work laptop.
### Steps to reproduce:
The following steps use `scons launcher` command but all the same applies to `scons dist` as well.
1. Get a fresh copy of NVDA:
```
> git clone --recursive https://github.com/nvaccess/nvda.git
> ...
> cd nvda
```
2. Build launcher:
```
> scons launcher
... many lines of actual building process omitted
scons: done building targets.
```
3. Check that launcher was built successfully:
```
> output\nvda_snapshot_source-master-6310b86.exe
```
4. Adding the following lines in core.py in def main right before `app.MainLoop()`:
```
import tones
tones.beep(1000, 1000)
```
5. Running NVDA from source to make sure beep can be heard:
```
> runnvda.bat
```
6. Rebuild launcher:
```
> scons launcher
...
scons: `launcher' is up to date.
scons: done building targets.
```
7. It appears scons didn't see any changes and didn't do anything, but let's try to run launcher again to see whether the beep can be heard:
```
$ output\nvda_snapshot_source-master-6310b86.exe
```

### Actual behavior:
`scons launcher` doesn't see code changes and does not rebuild after code change. After my attempt to rebuild, no beep is heard when launching launcher.
### Expected behavior:
`scons launcher` should be able to see code changes and rebuild launcher with code changes. Beep should be heard when launching launcher.

#### NVDA installed/portable/running from source:
Running from source
#### NVDA version:
Commit 6310b8661c7bc2f3d551fbe157a9e83a179e8b10
#### Windows version:
Windows 11
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

### PR DESCRIPTION
**Pull request by hwf1324**
### Link to issue number:

Fixes #13372

### Summary of the issue:

When rebuilding `dist`, some code changes were not detected. SCons does not trigger a new build.

### Description of user facing changes

None

### Description of development approach

`dist` will always be considered obsolete.

### Testing strategy:

Test steps from #13372:

> The following steps use `scons launcher` command but all the same applies to `scons dist` as well.
> 
> 1. Get a fresh copy of NVDA:
> 
> ```
> > git clone --recursive https://github.com/nvaccess/nvda.git
> > ...
> > cd nvda
> ```
> 
> 2. Build launcher:
> 
> ```
> > scons launcher
> ... many lines of actual building process omitted
> scons: done building targets.
> ```
> 
> 3. Check that launcher was built successfully:
> 
> ```
> > output\nvda_snapshot_source-master-6310b86.exe
> ```
> 
> 4. Adding the following lines in core.py in def main right before `app.MainLoop()`:
> 
> ```
> import tones
> tones.beep(1000, 1000)
> ```
> 
> 5. Running NVDA from source to make sure beep can be heard:
> 
> ```
> > runnvda.bat
> ```
> 
> 6. Rebuild launcher:
> 
> ```
> > scons launcher
> ...
> scons: `launcher' is up to date.
> scons: done building targets.
> ```

### Known issues with pull request:

It will always rebuild when building `dist`.

After running NVDA in the `dist` folder, encountered a Windows error during the rebuild. Speculating that py2exe is unable to replace the occupied file.
This issue, although not caused by this PR, should still be mentioned. See details in #15884.

There is no method found to only replace the modified files when rebuilding py2exe

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by codeofdusk** [comment]
This is intended behaviour, because `scons` only rebuilds on new commits.

**Comment by hwf1324** [comment]
> This is intended behaviour, because `scons` only rebuilds on new commits.

Yes, but see my comment in #13372 that SCons doesn't detect changes in dependencies that trigger a new build.

**Comment by lukaszgo1** [comment]
Would you be able to summarize (preferably with clear steps to reproduce) what are issues caused by this approach?

**Comment by hwf1324** [comment]
> Would you be able to summarize (preferably with clear steps to reproduce) what are issues caused by this approach?

Okay, I'll do that tomorrow.

For the case where NVDA in dist is run maybe it's not an issue for this PR.Maybe a new issue should be created.

**Comment by lukaszgo1** [comment]
Since #15884 has not been caused by this PR, I don't think this work should be blocked by it.
Ideally `dist` would only be rebuild when any file in the`source` directory changes, though if this is not easily doable this PR is certainly an improvement.
I'm not sure if there is an use case for building `dist` independently though. Currently the documentation states:
> To make a non-archived binary build (equivalent to an extracted portable archive), type: `scons dist`


however the created folder does not contain `nvda.exe`.
Can we consider not allowing building `dist` and change the default target to `launcher`?

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -167,6 +167,7 @@ Setting ``numCells`` is still supported for single line braille displays and ``n
   - Scripts that perform an action (e.g. move the cursor, change a parameter) should not speak in the "on-demand" mode.
   -
 - Fixed bug where deleting git-tracked files during `scons -c` resulted in missing UIA COM interfaces on rebuild. (#7070, #10833, @hwf1324)
+- Fix a bug where some code changes were not detected when building `dist` that prevented a new build from being triggered. Now always rebuilds every time `dist` is built. (#13372, @hwf1324)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:170]: The second sentence has a grammar mistake, does this work for you?

```suggestion
- Fix a bug where some code changes were not detected when building `dist`, that prevented a new build from being triggered. Now `dist` always rebuilds. (#13372, @hwf1324)
```

**Review by seanbudd — APPROVED** [review]
Thanks @hwf1324

**Pull request merged**